File: cosmoauthorization/src/java/it/csi/cosmo/cosmoauthorization/business/service/UseCaseService.java
/*
 * Copyright CSI-Piemonte - 2023
 * SPDX-License-Identifier: GPL-3.0-or-later
 */

package it.csi.cosmo.cosmoauthorization.business.service;

import it.csi.cosmo.cosmoauthorization.dto.rest.CategorieUseCaseResponse;
import it.csi.cosmo.cosmoauthorization.dto.rest.UseCaseResponse;

/**
 *
 */

public interface UseCaseService {

  /**
   * Restituisce una lista di , il cui numero di elementi e' pari al valore indicato da filter o, se
   * filter non e' valorizzato, un numero di elementi pari al valore definito nella tabella di
   * configurazione
   *
   * @param filter indica i filtri per la ricerca dei usecase
   * @return una lista di usecase
   */
  UseCaseResponse getUseCases(String filter);

  /**
   * Restituisce il usecase il cui id e' quello inserito nel database
   *
   * @param id del usecase da cercare sul database
   * @return il usecase del database
   */
  CategorieUseCaseResponse getUseCase(String id);

}

File: cosmowcl/src/app/shared/services/usecase.service.ts
/*
 * Copyright CSI-Piemonte - 2023
 * SPDX-License-Identifier: GPL-3.0-or-later
 */
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { CategorieUseCaseResponse } from '../models/api/cosmoauthorization/categorieUseCaseResponse';
import { UseCaseResponse } from '../models/api/cosmoauthorization/useCaseResponse';
import { ApiUrls } from '../utilities/apiurls';

@Injectable()
export class UseCaseService {

  constructor(
    private http: HttpClient
  ) {
    // NOP
  }

  /** GET: Restituisce uno UseCase. */
  getUseCases(filter: string): Observable<UseCaseResponse> {
    const options = filter ?
      { params: new HttpParams().set('filter', filter) } : {};
    return this.http.get<UseCaseResponse>(ApiUrls.USE_CASE, options);

  }

  /** GET: Restituisce una lista di categorie di UseCases. */
  getCategorieUseCase(codice: string): Observable<CategorieUseCaseResponse> {
    const options = codice ?
      { params: new HttpParams().set('string', codice) } : {};
    return this.http.get<CategorieUseCaseResponse>(ApiUrls.USE_CASE_CATEGORIE, options);

  }
}
